pebble/concurrent/process.py

```diff
@@ -17,16 +17,30 @@
 import os
 import types
 import multiprocessing
+import multiprocessing.context
 
 from itertools import count
 from functools import wraps
-from typing import Any, Callable
+from typing import Any, Callable, Optional, TypeVar, Union, overload
 from concurrent.futures import CancelledError, TimeoutError
 
 from pebble import common
 
+_P = TypeVar("_P")
```

Comment:
Same as here.

As this logic is common for both process and thread decorators, I would place it into `pebble.common.types` and use it around so we don't clutter the whole code base with backwards compatible supporting logic (try/excepts etc...).